### PR TITLE
TMEDIA 725: Convert Footer block

### PR DESCRIPTION
**Pull request by nschubach**
## Description

Convert Footer block to V2

## Jira Ticket

- [TMEDIA-725](https://arcpublishing.atlassian.net/browse/TMEDIA-725)

## Acceptance Criteria

Footer block is migrated to the new architecture
* Includes Custom fields of:
  * ‘Lazy Load block?’ checkbox - defaulted to false (with existing help text)\

## Test Steps

1. Checkout this branch `git checkout tmedia-725`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/footer-block`
3. Open `https://localhost/pagebuilder/pages`
4. Add Footer block to a test page
5. Configure content:
image
6. Run Storybook with `npm run storybook`
7. Check Footer Storybook

### After

image

## Dependencies or Side Effects

N/A

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/footer-block/features/footer/default.jsx [hunk 1]
```diff
@@ -41,7 +34,6 @@ const FooterItem = ({ customFields: { navigationConfig } }) => {
 		primaryLogoAlt,
 		locale = "en",
```
Comment (nschubach) [inline blocks/footer-block/features/footer/default.jsx]:
```suggestion
		locale,
```

File: blocks/footer-block/_index.scss [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		@include scss.block-properties("footer-links");
+	}
+
+	&__social-links {
+		&-container {
+			@include scss.block-components("footer-social-links-container");
+			@include scss.block-properties("footer-social-links-container");
+		}
+
+		a:not(:last-child) {
```
Comment (matthewroach) [inline blocks/footer-block/_index.scss]: This isn't used anywhere - is it still needed?

File: .storybook/themes/news.scss [hunk 3]
```diff
@@ -923,6 +924,86 @@
...
+						),
+						icon: (
+							fill: var(--color-primary),
+						),
+					),
+				),
+				footer-social-links-container: (
+					border-bottom: 1px solid var(--global-neutral-4),
+					width: 100%,
+					components: (
```
Comment (matthewroach) [inline .storybook/themes/news.scss:1046]: These appear to be duplicates of the values set on `footer-social-links`

**Review by matthewroach — CHANGES_REQUESTED** [review]

File: blocks/footer-block/_index.scss (hunk 2 above)
Comment (nschubach) [reply blocks/footer-block/_index.scss]: oh yeah.  I don't need that anymore.  Removing.

File: .storybook/themes/news.scss (hunk 3 above)
Comment (nschubach) [reply .storybook/themes/news.scss:1046]: Yep, I moved it and must have forgot to delete them.

**Review by matthewroach — DISMISSED** [review]

File: blocks/footer-block/README.md [hunk 4]
```diff
@@ -30,24 +30,16 @@ Available through `useContent`
 
 ## Internationalization fields
 
-- Add all internationalization fields used in the block
+- rss-link
+- facebook-link
+- twitter-link
```
Comment (ecarlisle) [inline blocks/footer-block/README.md]: We usually pair these in a markdown table with"default (English)" examples.

Comment (ecarlisle) [inline blocks/footer-block/README.md:35]: We usually pair these in a markdown table with"default (English)" examples.

File: blocks/footer-block/README.md [hunk 5]
```diff
@@ -30,24 +30,16 @@ Available through `useContent`
...
 
 ### Event Listening
 
-Include block specific instructions for event listening.
-
-OR
-
 This block does not emit any events.
 
 ## Additional Considerations
```
Comment (ecarlisle) [inline blocks/footer-block/README.md]:
```suggestion
```

Comment (ecarlisle) [reply blocks/footer-block/README.md]: We can probably take this out entirely.

**Review by ecarlisle — CHANGES_REQUESTED** [review]

File: blocks/footer-block/README.md [hunk 6]
```diff
@@ -30,24 +30,16 @@ Available through `useContent`
...
-| **eventName**  | Describe the event |
+n/a
 
 ### Event Listening
 
-Include block specific instructions for event listening.
-
-OR
-
 This block does not emit any events.
```
Comment (ecarlisle) [inline blocks/footer-block/README.md]:
```suggestion
This block does not emit or listen for events.
```

**Review by ecarlisle — APPROVED** [review]

File: blocks/footer-block/_index.scss [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+		@include scss.block-components("footer-links");
+		@include scss.block-properties("footer-links");
+	}
+
+	&__social-links {
+		&-container {
+			@include scss.block-components("footer-social-links-container");
+			@include scss.block-properties("footer-social-links-container");
+		}
+		> *:first-child {
```
Comment (nschubach) [inline blocks/footer-block/_index.scss:30]:
```suggestion

		> *:first-child {
```

File: blocks/footer-block/_index.scss [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+	&__social-links {
+		&-container {
+			@include scss.block-components("footer-social-links-container");
+			@include scss.block-properties("footer-social-links-container");
+		}
+
+		> *:first-child {
+			@include scss.block-components("footer-social-links-first");
+			@include scss.block-properties("footer-social-links-first");
```
Comment (matthewroach) [inline blocks/footer-block/_index.scss:32]: I see we are setting styles for last child but not first child. Is this for future proofing?

**Review by matthewroach — APPROVED** [review]

File: blocks/footer-block/_index.scss (hunk 8 above)
Comment (nschubach) [reply blocks/footer-block/_index.scss:32]: Yeah, I debated on whether I brought back the `not-last` token or exposing the first and last for styling.  I thought this was more flexible.

**Pull request merged**